zebra/rt_netlink.c

```diff
@@ -1124,8 +1105,11 @@ static int build_label_stack(struct mpls_label_stack *nh_label,
  * @param nexthop: Nexthop information
  * @param nlmsg: nlmsghdr structure to fill in.
  * @param req_size: The size allocated for the message.
+ *
+ * The function returns true if the nexthop could be added
+ * to the message, otherwise false is returned.
  */
-static void _netlink_route_build_singlepath(const struct prefix *p,
+static bool _netlink_route_build_singlepath(const struct prefix *p,
```

Comment:
can you say a bit about how this return status would be used? it seems as if there would already have been some data put into whereever 'nlmsg' is pointing - does that ... matter?

Reply:
This return status tells us if all attributes appended to the message in this function fits entirely in the buffer. This is used in `netlink_route_multipath_msg_encode`. And if I understand the second part of the question correctly, this function doesn't encode a separate message, but it adds some attributes to the existing one.

Reply:
that's sort of my question: if this returns 'false', it has already ... add some attributes (probably). how would a caller recover?

Reply:
I will make sure, but I think it doesn't matter whether the buffer is partially filled or not, because every time we encode a new message we firstly fill it with zeroes.